File: src/main/java/com/curseclient/client/events/CanRenderInLayerEvent.java
package com.curseclient.client.events;

import net.minecraft.block.Block;
import net.minecraft.util.BlockRenderLayer;

public class CanRenderInLayerEvent {
    private BlockRenderLayer blockRenderLayer;
    private final Block block;

    public CanRenderInLayerEvent(final Block block) {
        this.blockRenderLayer = null;
        this.block = block;
    }

    public BlockRenderLayer getBlockRenderLayer() {
        return blockRenderLayer;
    }

    public void setBlockRenderLayer(BlockRenderLayer blockRenderLayer) {
        this.blockRenderLayer = blockRenderLayer;
    }

    public Block getBlock() {
        return block;
    }

}

File: src/main/kotlin/com/curseclient/client/event/events/render/CanRenderInLayerEvent.kt
package com.curseclient.client.event.events.render

import net.minecraft.block.Block
import net.minecraft.util.BlockRenderLayer

// I HATE JVM
class CanRenderInLayerEvent(val block: Block) {
    var blockRenderLayer: BlockRenderLayer? = null

    fun gettingBlockRenderLayer(): BlockRenderLayer? {
        return blockRenderLayer
    }

    fun setUpBlockRenderLayer(blockRenderLayer: BlockRenderLayer) {
        this.blockRenderLayer = blockRenderLayer
    }

    fun gettingBlock(): Block {
        return block
    }
}
